### DISCUSSION TITLE
Share code between setup_suite and tests

### DISCUSSION THREAD

**Discussion by jkroepke** [bats-core/bats-core#636] [Q&A] [answered]
Hi,

I have multiple helper functions that I share between tests and setup_file functions.
Since `setup_suite` is now official available, I consider to move some setup stuff from `setup_file` to `setup_suite`.

Currently, I'm unable to load additional files through `load` on `setup_suite.bash`

```
% ./tests/bats/core/bin/bats -r ./tests/unit/template.bats
/Users/jkr/IdeaProjects/private/helm-secrets/tests/unit/setup_suite.bash: line 3: load: command not found
 ✗ setup_suite
   bats warning: Executed 1 instead of expected 59 tests

59 tests, 1 failure, 58 not run
```

Do I have to define all the helper functions twice? Is it possible to load helper functions on setup_suite.bash once and have them available for the whole suite?

Instead pollute the issues here with my question, I feel open a discussion here is a better approch.

**Comment by martin-schulze-vireso** [comment, marked as answer]
Unless the helper functions are exported, you will have to load them multiple Times, once per process. Setup_suite, setup_file and setup/test have their own process, so three in total.

The load function not being available in setup_suute is an oversight. Please Open an issue for that.

**Reply by jkroepke** [reply]
https://github.com/bats-core/bats-core/issues/637

**Reply by jkroepke** [reply]
Currently, I have a helper.bash file defined which includes helper functions and `setup_file`. That could be the reason, why I do not have the export them multiple times.
